scipy/signal/_signaltools.py

```diff
@@ -1569,7 +1569,7 @@ def medfilt(volume, kernel_size=None):
     numels = np.prod(kernel_size, axis=0)
     order = numels // 2
 
-    if volume.dtype in [np.bool_, np.cfloat, np.cdouble, np.clongdouble,
+    if volume.dtype in [np.bool_, np.complex128, np.clongdouble,
```

Comment:
also correct as a 1-for-1 replacement. I had to do a double take here - `cfloat` is not `complex64` as one would expect, but also `complex128`. 

That said, I think this code had a bug - it seems clear that the author intended to write `np.complex64` here (which is `np.csingle` rather than `np.cfloat`). Currently we get:
```python
>>> import numpy as np
>>> from scipy import signal
>>> x = np.linspace(0, 1, dtype=np.complex64)
>>> signal.medfilt(x)
File ~/code/scipy/build-install/lib/python3.10/site-packages/scipy/ndimage/_filters.py:1411, in _rank_filter(input, rank, size, footprint, output, mode, cval, origin, operation, axes)
   1409 input = numpy.asarray(input)
   1410 if numpy.iscomplexobj(input):
-> 1411     raise TypeError('Complex type not supported')
   1412 axes = _ni_support._check_axes(axes, input.ndim)
   1413 num_axes = len(axes)

TypeError: Complex type not supported
```

And after adding `np.complex64` to the list we'd get:
```python
  File ~/code/scipy/build-install/lib/python3.10/site-packages/scipy/signal/_signaltools.py:1574 in medfilt
    raise ValueError(f"dtype={volume.dtype} is not supported by medfilt")
ValueError: dtype=complex64 is not supported by medfilt
```

I think it'd be a valid bug fix to add `complex64` here.

Reply:
That's clear now - I changed used dtypes to sized aliases (both in `medfilt`'s implementation and test).